### PR TITLE
Add docs for iceberg expire snapshots and delete orphan files

### PR DESCRIPTION
**Pull request by homar**
## Description
fixes https://github.com/trinodb/trino/issues/12086
## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -201,6 +201,42 @@ to the filter:
...
+
+expire_snapshots
+""""""""""""""""
+
+The ``expire_snapshots`` command removes all snapshots and all related metadata and data files.
+Regularly expiring snapshots is recommended to delete data files that are no longer needed,
+and to keep the size of table metadata small.
+The age of snapshots that will be expired is defined by ``retention_threshold`` parameter.
+
+``EXPIRE_SNAPSHOTS`` can be run as follows:
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:213]: lowercase

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -201,6 +201,42 @@ to the filter:
...
+this property is ``7d``.
+
+delete_orphan_files
+"""""""""""""""""""
+
+The ``delete_orphan_files`` command removes all files from table's data directory which are
+not linked from metadata files and that are older than the value of ``retention_threshold`` parameter.
+Deleting orphan files from time to time is recommended to keep size of table's data directory under control.
+
+``DELETE_ORPHAN_FILES`` can be run as follows:
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:230]: lowercase

**Review by findepi — APPROVED** [review]

**Review by findepi — APPROVED** [review]

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -201,6 +201,41 @@ to the filter:
     ALTER TABLE test_partitioned_table EXECUTE optimize
     WHERE partition_key = 1
 
+expire_snapshots
+~~~~~~~~~~~~~~~~
+
+The ``expire_snapshots`` command removes all snapshots and all related metadata and data files.
+Regularly expiring snapshots is recommended to delete data files that are no longer needed,
+and to keep the size of table metadata small.
+The age of snapshots that will be expired is defined by ``retention_threshold`` parameter.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:210]: The procedure affects all snapshots that are older than the time period configured with the ``retention_threshold`` parameter.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -201,6 +201,41 @@ to the filter:
...
+
+``expire_snapshots`` can be run as follows:
+
+.. code-block:: sql
+
+  ALTER TABLE test_table EXECUTE expire_snapshots(retention_threshold => '7d')
+
+The ``iceberg.expire_snapshots.min-retention`` config property provides a safety
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:218]: drop "config" or use "catalog configuration property" ..

also probably reword the sentence to be specific .. "safety measure" doesnt really spell out what happens .. I think this is the case...  snapshots within min-retention time period are not affected by the procedure..  its basically an override .. 

example scenario: min-retention set to 30 days

call with retention_threshold 21 days
snapshots from 30 days ago and older are deleted .. 

call with 30 days
nothing happens

call with 60 days
snapshots from 60 days ago and older are deleted ..

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 5]
```diff
@@ -201,6 +201,41 @@ to the filter:
...
+measure to ensure that files are retained as expected.  The default value for
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:219]: remove space before "The"

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 6]
```diff
@@ -201,6 +201,41 @@ to the filter:
...
+not linked from metadata files and that are older than the value of ``retention_threshold`` parameter.
+Deleting orphan files from time to time is recommended to keep size of table's data directory under control.
+
+``delete_orphan_files`` can be run as follows:
+
+.. code-block:: sql
+
+  ALTER TABLE test_table EXECUTE delete_orphan_files(retention_threshold => '7d')
+
+The ``iceberg.delete_orphan_files.min-retention`` config property provides a safety
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:235]: Similar feedback to above

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (homar) [reply docs/src/main/sphinx/connector/iceberg.rst:218]: nope, you just can't run the procedure with retention_threshold smaller than iceberg.expire_snapshots.min-retention

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst]: ok .. well.. then we need to clarify that in the docs. 

And also .. is there an useful error message when you try?

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 7]
```diff
@@ -201,6 +201,41 @@ to the filter:
...
+and to keep the size of table metadata small.
+The procedure affects all snapshots that are older than the time period configured with the ``retention_threshold`` parameter.
+
+``expire_snapshots`` can be run as follows:
+
+.. code-block:: sql
+
+  ALTER TABLE test_table EXECUTE expire_snapshots(retention_threshold => '7d')
+
+The ``iceberg.expire_snapshots.min-retention`` catalog configuration property prevents from running
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:218]: This sentence is too convoluted . how about

```
The value for ``retention_threshold`` must be higher than ``iceberg.expire_snapshots.min-retention`` in the catalog.
```

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (homar) [reply docs/src/main/sphinx/connector/iceberg.rst]: yes there is an useful message but it is quite long: `Retention specified (33.00s) is shorter than the minimum retention configured in the system (7.00d). Minimum retention can be changed with iceberg.expire_snapshots.min-retention configuration property or iceberg.expire_snapshots_min_retention session property`

**Review by mosabua — APPROVED** [review]
Ideally we wrap the source at 80 char but its more important to get this merged ..

**Comment by mosabua** [comment]
@martint could we get this in for 379 .. feature shipped in 378 already..

**Pull request merged**